crypto/src/hash.rs

```diff
@@ -27,36 +28,34 @@ ffi::ffi_item! {
         PartialEq,
         Ord,
         PartialOrd,
-        Decode,
-        Encode,
-        Deserialize,
-        Serialize,
         IntoSchema,
         FfiType,
     )]
-    #[repr(transparent)]
-    #[serde(transparent)]
     // TODO: use #[ffi_type(unsafe {robust})] instead
     #[ffi_type(opaque)]
-    #[display(fmt = "{}", "hex::encode(_0)")]
-    #[debug(fmt = "{{ Hash({}) }}", "hex::encode(_0)")]
-    pub struct Hash([u8; Self::LENGTH]);
+    #[display(fmt = "{}", "hex::encode(self.as_ref())")]
+    #[debug(fmt = "{{ Hash({}) }}", "hex::encode(self.as_ref())")]
+    #[repr(C)]
+    pub struct Hash {
+        bytes: [u8; Self::LENGTH - 1],
+        byte: NonZeroU8,
```

Comment:
These names are tolerable. But not the best. Given that the things are completely internal I'd propose calling them `more_significant_bits`, and `least_significant_byte`.